airframe-fluentd/src/main/scala/wvlet/airframe/fluentd/FluencyClient.scala

```diff
@@ -14,28 +14,39 @@
 package wvlet.airframe.fluentd
 import java.time.Instant
 
+import org.komamitsu.fluency.fluentd.FluencyBuilderForFluentd
+import org.komamitsu.fluency.ingester.sender.ErrorHandler
 import org.komamitsu.fluency.{EventTime, Fluency}
 import wvlet.log.LogSupport
 import wvlet.airframe._
 
-case class FluencyConfig(
+case class FluencyClientConfig(
+    host: String = "127.0.0.1",
+    port: Int = 24224,
     // Use the extended EventTime timestamps
     // https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1#eventtime-ext-format
     useExtendedEventTime: Boolean = false,
-    fluencyConfig: Fluency.Config = new Fluency.Config()
+    maxBufferSize: Long = 512 * 1024 * 1024,
+    flushIntervalMillis: Int = 600,
+    jvmHeapBufferMode: Boolean = true,
+    ackResponseMode: Boolean = true,
+    sslEnabled: Boolean = true,
+    fileBackupDir: String = null,
+    errorHandler: ErrorHandler = null
 )
 
-object FluencyClient {
-  def newFluency(fluentdConfig: FluentdConfig, fluencyConfig: FluencyConfig): Fluency = {
-    Fluency.defaultFluency(fluentdConfig.host, fluentdConfig.port, fluencyConfig.fluencyConfig)
-  }
-}
-
 trait FluencyClient extends FluentdClient with LogSupport {
-  private val fluentdConfig = bind[FluentdConfig]
-  private val fluencyConfig = bind[FluencyConfig]
-  private val fluency: Fluency = bind { (fluentdConfig: FluentdConfig, fluencyConfig: FluencyConfig) =>
-    FluencyClient.newFluency(fluentdConfig, fluencyConfig)
+  private val fluencyClientConfig = bind[FluencyClientConfig]
+  private val fluency: Fluency = bind { (fluencyClientConfig: FluencyClientConfig) =>
+    val builder = new FluencyBuilderForFluentd()
+    builder.setMaxBufferSize(fluencyClientConfig.maxBufferSize)
+    builder.setFlushIntervalMillis(fluencyClientConfig.flushIntervalMillis)
+    builder.setJvmHeapBufferMode(fluencyClientConfig.jvmHeapBufferMode)
+    builder.setAckResponseMode(fluencyClientConfig.ackResponseMode)
+    builder.setSslEnabled(fluencyClientConfig.sslEnabled)
+    builder.setFileBackupDir(fluencyClientConfig.fileBackupDir)
+    builder.setErrorHandler(fluencyClientConfig.errorHandler)
```

Comment:
Does it accept null values?

Reply:
Yes, builder's default value is null. If it's null, builder doesn't do anything.